### ISSUE TITLE
No rule checking for redefinition of variables

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#909]
The following code:

``` js
if (true) {
    var a = 2;
} else {
    var a = 3;
}
```

makes jsHint output: `W004: 'a' is already defined.` but ESLint doesn't complain and there seems to be no rule to enforce this check.

Note that the rule has to take into account that `var` has function scope and `const`/`let` have block scope so above code with `s/var/const` shouldn't trigger any errors/warnings.
